openmeter/customer/adapter/customer.go

```diff
@@ -291,14 +293,42 @@ func (a *adapter) GetCustomer(ctx context.Context, input customer.GetCustomerInp
 				WithSubscription(func(query *entdb.SubscriptionQuery) {
 					applyActiveSubscriptionFilter(query, clock.Now().UTC())
 					query.WithPlan()
-				}).
-				Where(customerdb.ID(input.ID)).
-				Where(customerdb.Namespace(input.Namespace))
+				})
+
+			if input.CustomerID != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerID.Namespace))
+				query = query.Where(customerdb.ID(input.CustomerID.ID))
+			} else if input.CustomerKey != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerKey.Namespace))
+				query = query.Where(customerdb.Key(input.CustomerKey.Key))
+			} else if input.CustomerIDOrKey != nil {
+				query = query.Where(customerdb.Namespace(input.CustomerIDOrKey.Namespace))
+				query = query.Where(customerdb.Or(
+					customerdb.ID(input.CustomerIDOrKey.IDOrKey),
+					customerdb.Key(input.CustomerIDOrKey.IDOrKey),
+				))
```

Comment:
Please:
-  filter for deletedAt != nil or we bring back the old bug of returning random deleted entities by key

Reply:
Added